### PR TITLE
Disable join types in validators

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Disable join types in validators
- This will also fix the build issue due to grammar file change
Related: https://github.com/opensearch-project/sql/pull/3039

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: async-query-core/src/main/java/org/opensearch/sql/spark/validator/S3GlueGrammarElementValidator.java [hunk 1]
```diff
@@ -50,6 +55,11 @@ public class S3GlueGrammarElementValidator extends DenyListGrammarElementValidat
               HINTS,
               INLINE_TABLE,
               FILE,
+              CROSS_JOIN,
+              LEFT_SEMI_JOIN,
+              RIGHT_OUTER_JOIN,
+              FULL_OUTER_JOIN,
+              LEFT_ANTI_JOIN,
```
Comment (dai-chen) [inline async-query-core/src/main/java/org/opensearch/sql/spark/validator/S3GlueGrammarElementValidator.java:62]: These JOIN types are disallowed even for S3 data source?

Comment (ykmr1224) [reply async-query-core/src/main/java/org/opensearch/sql/spark/validator/S3GlueGrammarElementValidator.java:62]: Yes, limiting regardless of the data source.

Comment (dai-chen) [reply async-query-core/src/main/java/org/opensearch/sql/spark/validator/S3GlueGrammarElementValidator.java:62]: So we only limit these JOIN types due to performance/cost? Others like INNER JOIN, LEFT JOIN can still work?

Comment (ykmr1224) [reply async-query-core/src/main/java/org/opensearch/sql/spark/validator/S3GlueGrammarElementValidator.java:62]: Yes, intention is only allowing INNER JOIN and LEFT OUTER JOIN.

**Review by noCharger — APPROVED** [review]
thx for the change

**Pull request merged**